### ISSUE TITLE
[Firefox] Focus is lost after navigating to Dialogs in full screen mode

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/john-travoltage#252]
Found by @phet-steele for For phetsims/tasks#838.

Steps to reproduce:
- Enter full screen mode
- navigate to the Keyboard Help Dialog or the Updates Dialog.
- Tab away from the "close" button.
- Focus will be lost, and we won't be able to focus anything in the sim until using mouse or touch

Reproduced in Firefox on Windows 8.1 and Windows 7.

**Comment by jessegreenberg** [comment]
@phet-steele recommended that we use NVDA to hear what is happening to focus.

After turning on NVDA, the problem went away on Windows 8.1.  Now we can no longer reproduce the problem on that machine, even when NVDA is off.

Still happening on Windows 7 Firefox.

**Comment by jessegreenberg** [comment]
I added this bit of code to track the active element when pressing tab, I will take a look at what is happening.
```js
    document.addEventListener( 'keyup', function( event ) {
      if ( event.keyCode === 9 ) {
        console.log( document.activeElement );
      }
    } );
```

But I am starting to wonder if tabbing out of the button is causing the window to lose focus, which means no keyup events to track.

**Comment by jessegreenberg** [comment]
Also, this is not happening at all for the About Dialog. What is different in that case?  More than one focusable element?

**Comment by jessegreenberg** [comment]
Note: If we can't figure out why this is happening, we could potentially limit focus to the Dialog, so it is impossible to tab out of the Dialog. This might be bad for a11y, I am not sure.

**Comment by jessegreenberg** [comment]
I believe that this is the bug we are seeing:
https://bugzilla.mozilla.org/show_bug.cgi?id=910136

**Comment by jessegreenberg** [comment]
That bug report is marked as resolved only because communication stalled.

**Comment by jessegreenberg** [comment]
When shift tabbing, focus in full screen Firefox looks like

- ```<html>```
- ```<body>```
- '''<button>``` (the close button)

When tabbing, focus looks like
- ```<body>```
- ```<body>```
- ```<body>```
- ```<body>```
- ```<body>```
- ```<body>```
...
Focus is stuck on body.

Maybe we can have a check that says if focus is about to be placed on the body, refrain from moving focus?

**Comment by jessegreenberg** [comment]
If I add one more focusable thing to the dialog, things work just fine. This seems like a Firefox bug where having only one focusable element in the DOM breaks things when the nav bar is hidden.

**Comment by jessegreenberg** [comment]
That should mean we can try to detect fullscreen mode, and whether or not the document has any more focusable elements.  If not, we can prevent default behavior on tab.

**Comment by jessegreenberg** [comment]
Fixed via https://github.com/phetsims/joist/commit/cc645a0d12179c546f32b4df0397b8407c0459e3.

**Comment by phet-steele** [comment]
Looks good, thanks.

**Comment by jessegreenberg** [comment]
This bug has resurfaced in https://github.com/phetsims/QA/issues/66.

**Comment by jessegreenberg** [comment]
Fixed in master, patch applied to  the 1.4 releases branch of RIAW.

**Comment by phet-steele** [comment]
RIAW checks out on Win 7 FFQ.
